lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -1107,10 +1159,73 @@ static ValueDecl *deriveDecodable_init(TypeChecker &tc, Decl *parentDecl,
 ///
 /// \param target The type to validate.
 ///
+/// \param requirement The requirement we want to synthesize.
+///
 /// \param proto The *codable protocol to check for validity.
 static bool canSynthesize(TypeChecker &tc, NominalTypeDecl *target,
-                          ProtocolDecl *proto) {
-  // First, look up if the type has a valid CodingKeys enum we can use.
+                          ValueDecl *requirement, ProtocolDecl *proto) {
+  // Before we attempt to look up (or more importantly, synthesize) a CodingKeys
+  // entity on target, we need to make sure the type is otherwise valid.
+  //
+  // If we are synthesizing Decodable and the target is a class with a
+  // superclass, our synthesized init(from:) will need to call either
+  // super.init(from:) or super.init() depending on whether the superclass is
+  // Decodable itself.
+  //
+  // If the required initializer is not available, we shouldn't attempt to
+  // synthesize CodingKeys.
```

Comment:
@jrose-apple I don't think that's problematic — we can always add a private type to a subclass with the same name as a superclass's public type:

```swift
public class Super {
    public enum X {
        case somePublic
    }
}

public class Sub : Super {
    private enum X {
        case myPrivate
    }
}
```

^ works just fine
We always synthesize a private enum, so I don't think this is problematic.